lib/artemis_api/client.rb

```diff
@@ -16,48 +17,48 @@ def initialize(access_token, refresh_token, expires_in, created_at, options = {}
 
       @oauth_client = OAuth2::Client.new(@options[:app_id], @options[:app_secret], site: @options[:base_uri])
       @oauth_token = OAuth2::AccessToken.from_hash(
-                      oauth_client, 
+                      oauth_client,
                       {access_token: @access_token,
                        refresh_token: @refresh_token,
                        created_at: @created_at,
                        expires_in: @expires_in,
                        expires_at: @expires_at})
-      @objects = {} 
+      @objects = {}
     end
 
-    def find_one(type, id, force = false)
+    def find_one(type, id, facility_id: nil, include: nil, force: false)
       obj = get_record(type, id)
       if !obj || force
-        response = @oauth_token.get("#{@options[:base_uri]}/api/v3/#{type}/#{id}")
-        if response.status == 200
-          json = JSON.parse(response.body)
-          obj = store_record(type, id, json['data'])
-
-          #if json['included']
-          #  json['included'].each do |included_obj|
-          #    store_record(included_obj['type'], included_obj['id'], included_obj)
-          #  end
-          #end
-        end
+        refresh if @oauth_token.expired?
+
+        url = if facility_id
```

Comment:
this approach doesn't seem like it'll scale well to support the other APIs (like `tank_readings` or `sensor_data` that are nested even further, but we can cross that bridge that bridge when we come to it.